File: Crio/crio_programming_interview_problems-master/RemoveDuplicatesFromSortedArray/RemoveDuplicatesFromSortedArray.cpp
#include <bits/stdc++.h>
#include "../crio/cpp/io/FastIO.hpp"
#include "../crio/cpp/io/ReadMatrix.hpp"

using namespace std;

class RemoveDuplicatesFromSortedArray {
public:

    int removeDuplicates(vector<int>& nums) {
        int ans = 0;
        // CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
        int n = nums.size();
        for (int i = 0; i < n; ) {
            int x = nums[i];
            nums[ans++] = nums[i++];
            while (i < n && nums[i] == x) {
                i++;
            }
        }
        // CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
        return ans;
    }
};

int main() {
    FastIO();
    int n;
    cin >> n;
    vector<int> nums;
    ReadMatrix<int>().OneDMatrix(n, nums);
    int result = RemoveDuplicatesFromSortedArray().removeDuplicates(nums);
    cout << result;
    return 0;
}

File: Crio/crio_programming_interview_problems-master/RemoveDuplicatesFromSortedArray/RemoveDuplicatesFromSortedArray.py

def removeDuplicates(nums):
   
    # CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
    n=len(nums)
    if n<=1:
        return n
    cnt=0
    i=0
    while i<n:
        if nums[i]==nums[cnt]:
            i+=1
        else:
            cnt+=1
            nums[cnt]=nums[i]
        
    return cnt+1
    # CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

    return 0

if __name__ == '__main__':
    n = int(input())
    nums = []
    if n > 0:
        nums = input().split()
    nums = [int(i) for i in nums]
    result = removeDuplicates(nums)
    print(result)
